src/OpenTelemetry.Instrumentation.Dependencies/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -72,12 +76,17 @@ public override void OnCustom(string name, Activity activity, object payload)
                         var connection = this.connectionFetcher.Fetch(command);
                         var database = this.databaseFetcher.Fetch(connection);
 
+                        // SqlClient does not create an Activity. So the activity coming in here will be null or the root span.
+                        activity = SqlClientActivitySource.StartActivity(ActivityName, ActivityKind.Client);
```

Comment:
Since `database` is known at this stage, I'd use that as name instead of `ActivityName` and then update later.
Or alternately, move this code to even before we attempt to fetch anything from payload. - I think this approach is the correct one as we wont waste cycles to Fetch, unless activity is sampled-in.

Reply:
I went with option 2, short-circuiting everything if we don't decide to sample. Great for perf. The only downside I could think of is the sampler will have little to no data about the Activity if someone wanted to make a decision based on the database name or something.